applications/StructuralMechanicsApplication/custom_conditions/line_load_condition.cpp

```diff
@@ -0,0 +1,458 @@
+
+    for ( IndexType i = 0; i < number_of_nodes; i++ ) {
+        const IndexType row_index = i * TDim;
+
+        for ( IndexType j = 0; j < number_of_nodes; j++ ) {
+            const IndexType col_index = j * TDim;
+
+            const double coeff = Pressure * rN[i] * rDN_De( j, 0 ) * IntegrationWeight;
+            noalias(Kij) = coeff * cross_tangent_xi;
+
+            MathUtils<double>::AddMatrix( rK, Kij, row_index, col_index );
+        }
+    }
+
+    KRATOS_CATCH( "" )
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<std::size_t TDim>
+void LineLoadCondition<TDim>::CalculateAndAddPressureForce(
+    Vector& rRightHandSideVector,
+    const Vector& rN,
+    const array_1d<double, 3>& rNormal,
+    double Pressure,
+    double IntegrationWeight
+    ) const
+{
+    const SizeType number_of_nodes = this->GetGeometry().size();
+    const SizeType block_size = this->GetBlockSize();
+
+    for ( IndexType i = 0; i < number_of_nodes; ++i ) {
+        const IndexType index = block_size * i;
+
+        const double coeff = Pressure * rN[i] * IntegrationWeight;
+
+        for ( IndexType j = 0; j < TDim; ++j ) {
+            rRightHandSideVector[index + j] -= coeff * rNormal[j];
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+void LineLoadCondition<2>::GetLocalAxis1(
+    array_1d<double, 3>& rLocalAxis,
+    const Matrix& rJacobian
+    ) const
+{
+    rLocalAxis[0] = rJacobian(0, 0);
+    rLocalAxis[1] = rJacobian(1, 0);
+    rLocalAxis[2] = 0.0;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+void LineLoadCondition<3>::GetLocalAxis1(
+    array_1d<double, 3>& rLocalAxis,
+    const Matrix& rJacobian
+    ) const
+{
+    rLocalAxis[0] = rJacobian(0, 0);
+    rLocalAxis[1] = rJacobian(1, 0);
+    rLocalAxis[2] = rJacobian(2, 0);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+void LineLoadCondition<2>::GetLocalAxis2(array_1d<double, 3>& rLocalAxis) const
+{
+    rLocalAxis[0] = 0.0;
+    rLocalAxis[1] = 0.0;
+    rLocalAxis[2] = 1.0;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+void LineLoadCondition<3>::GetLocalAxis2(array_1d<double, 3>& rLocalAxis) const
+{
+    KRATOS_ERROR_IF(!Has(LOCAL_AXIS_2)) << "The variable LOCAL_AXIS_2 is needed to compute the normal" << std::endl;
```

Comment:
what happens if we have a torque?

Reply:
We can add that in the future, ask to @armingeiser how it is intended

Reply:
@KlausBSautter

Reply:
shouldn't this be calculated by the element? e.g. the beam has a post-processing feature of calculating the **current** axis2
in `void CrBeamElement3D2N::CalculateOnIntegrationPoints(
    const Variable<array_1d<double, 3>>& rVariable,
    std::vector<array_1d<double, 3>>& rOutput`